src/udev/scsi_id/scsi_id.c

```diff
@@ -180,9 +181,14 @@ static int get_file_options(const char *vendor, const char *model,
         for (;;) {
                 vendor_in = model_in = options_in = NULL;
 
-                buf = fgets(buffer, MAX_BUFFER_LEN, f);
-                if (buf == NULL)
+                r = read_line(f, MAX_BUFFER_LEN, &buf);
```

Comment:
This change means we now read everything into a buffer 'buf'. That buffer is never freed (it needs to be freed on every line). Moreover the buffer allocated as `buffer` isn't useful anymore and should be removed